src/main/kotlin/calculator/CalculatorInputParser.kt

```diff
@@ -0,0 +1,60 @@
+package calculator
+
+import util.isEven
+import util.isOdd
+
+class CalculatorInputParser {
```

Comment:
p2; parser와 validator로 분리한 것 너무 좋습니다. 👍
더 나아가서 메소드 기반의 클래스가 아닌 객체에게 책임과 역할을 준다면 어떨까요?
nullable한 input도 아래(CalculatorInput의 예시)와 같은 형태로 만든다면 !! 키워드를 사용하는 부분도 해결할 수 있을 거 같아요. 🙂

```suggestion
class CalculatorInput(
    val numbers: List<Double>,
    val operators: List<Operator>,
) {

    init {
        require(value = 오퍼레이션과 숫자의 개수 검증 + operator !in fourBasicOperators) {
           ~
        }
    }
```

Reply:
서비스가 아닌 도메인 위주로 로직을 풀어나가는 방식이군요.
이 방식으로도 구현해보고 비교해보면 좋을 것 같아 익숙치는 않지만 한번 구현해볼게요!

Reply:
일반적으로 서비스 레이어에서 작성하는 로직과 다르게 객체지향 프로그래밍으로 생각해 주시면 더욱 쉽게 접근하실 수 있을 거 같아요!